single precision support in skimage.filters

grlee77 opened the pull request:
## Description

This PR contains the commits from #5353, so that one should be reviewed prior to this (float32 support Hessian functions in skimage.feature were required in the ridge filters implementation here).


## Checklist

<!-- It's fine to submit PRs which are a work in progress! -->
<!-- But before they are merged, all PRs should provide: -->
- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

<!-- For detailed information on these and other aspects see -->
<!-- the scikit-image contribution guidelines. -->
<!-- https://scikit-image.org/docs/dev/contribute.html -->

## For reviewers

<!-- Don't remove the checklist below. -->
- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

pep8speaks commented:
Hello @grlee77! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:



























































* In the file [`skimage/filters/tests/test_lpi_filter.py`](https://github.com/scikit-image/scikit-image/blob/0a64d60973577396f791d7e774f4da3c57e51548/skimage/filters/tests/test_lpi_filter.py):

> [Line 13:1](https://github.com/scikit-image/scikit-image/blob/0a64d60973577396f791d7e774f4da3c57e51548/skimage/filters/tests/test_lpi_filter.py#L13): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1

rfezzani commented on skimage/filters/lpi_filter.py:
```diff
@@ -4,15 +4,16 @@
 def _min_limit(x, val=eps):
-    mask = np.abs(x) < eps
-    x[mask] = np.sign(x[mask]) * eps
+    mask = np.abs(x) < val
```
:eyes: good catch :clap:

rfezzani commented on skimage/filters/lpi_filter.py:
```diff
@@ -195,7 +200,7 @@ def inverse(data, impulse_response=None, filter_params={}, max_gain=2,
 
     F, G = filt._prepare(data)
-    _min_limit(F)
+    _min_limit(F, val=np.finfo(float).eps)
```
```suggestion
    _min_limit(F, val=eps)
```
`eps` is already defined as a global variable. But I am not against your writing, in this case I suggest removing the global declaration of `eps`, just for consistency :wink:

rfezzani commented on skimage/filters/lpi_filter.py:
```diff
@@ -238,7 +243,7 @@ def wiener(data, impulse_response=None, filter_params={}, K=0.25,
 
     F, G = filt._prepare(data)
-    _min_limit(F)
+    _min_limit(F, val=np.finfo(float).eps)
```
Same here

rfezzani reviewed (approved):
Excellent, thank you @grlee77.

grlee77 commented:
Hi @rfezzani, thanks for reviewing this previously. I fixed conflicts and addressed your comment about the global `eps` variable.

Since the last time you reviewed this, I added single precision support to the three `threshold_*` functions that return arrays (see commits 42c9a94 and 5ebcf52).  

Also, it could go in a separate standalone PR if preferred, but there are a couple of minor fixes to prior single precision PRs in fbb1df9 (a dtype, not an image should be passed to `_supported_float_type`).

rfezzani commented:
Thank you @grlee77, a second approval and we are good to go :wink: !

grlee77 commented:
This is the only one of the single precision PRs that has not yet been merged.  It has one approval now and the changes made here are consistent with the already merged PRs in this area. Given that, I would be inclined to merge it as is rather than ask others to spend substantial time doing additional reviews. Are there any objections to this?

rfezzani commented:
> ... Are there any objections to this?

Not from my side ^^
